docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,663 @@
+        response: JSON.stringify({ chainATransaction, chainBTransaction }),
+      });
+    };
+```
+
+As everything on the blockchain works on hexadecimal hashes, we need to hash our transactions as well.
+
+```js
+    const hashTransaction = (tx) => {
+      return ethers.utils.arrayify(
+        ethers.utils.keccak256(
+          ethers.utils.arrayify(ethers.utils.serializeTransaction(tx)),
+        ),
+      );
+    }
+
+    Lit.Actions.setResponse({ response: "Conditions for swap not met!" });
+  }
+go();
+```
+
+We need to remember to set `Lit.Actions.setResponse` before we finish executing the Lit Action. This defines the response that will be shown when the Lit Action finishes executing.
+
+The complete swap Lit Action can be found [here](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapAction.js).
+
+## 3) Implementing Mint/Grant/Burn with SDK
+
+### Installing and importing the relevant packages
+
+```bash
+npm i @lit-protocol/lit-node-client @lit-protocol/contracts-sdk 
+@simplewebauthn/browser  @lit-protocol/types ethers
+```
+
+
+```js
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { LitContracts } from "@lit-protocol/contracts-sdk";
+import { LitNetwork, AuthMethodScope, LIT_CHAINS } from "@lit-protocol/constants";
+import { LitAbility } from "@lit-protocol/types";
+import {
+    LitActionResource,
+    createSiweMessageWithRecaps,
+    generateAuthSig,
+    LitPKPResource,
+} from "@lit-protocol/auth-helpers";
+import { ethers } from "ethers";
+import bs58 from "bs58";
+```
+
+### Creating our two wallet instances
+
+These wallets will represent the two parties involved in the swap.
+
+```js
+async function getWalletA() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_1,
+        provider
+    );
+    return wallet;
+}
+
+async function getWalletB() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_2,
+        provider
+    );
+    return wallet;
+}
+```
+
+### Creating our Lit Action and uploading it to IPFS
+
+Once we are done with our Lit Action, we need to upload it to IPFS so we get an immutable CID which will always point to our action.
+
+```js
+export async function createLitAction() {
+    console.log("creating lit action..");
+    const action = createERC20SwapLitAction(chainAParams, chainBParams);
+    const ipfsCid = await uploadViaPinata(action);
+
+    console.log("Lit Action code:\n", action);
+    console.log("IPFS CID: ", ipfsCid);
+    return ipfsCid;
+}
+```
+
+### Create a Mint/Grant/Burn PKP
+
+:::info
+Currently, direct calls to the smart contract’s m/g/b function are currently failing with the Lit Network. We’ll replicate the same functionality using the SDK.
```

Comment:
> Currently, direct calls to the smart contract’s m/g/b function are currently failing with the Lit Network. We’ll replicate the same functionality using the SDK.

"The M/G/B function is currently broken. For this example, we'll create a workaround the replicates the same functionality using the Lit SDK"

Reply:
Why is this currently broken? Do we have an ETA for a fix?

Reply:
Not exactly, Adarsh has been working on the issue earlier. Check [LIT-3762](https://linear.app/litprotocol/issue/LIT-3762/cant-authenticate-for-burnt-pkp).

Reply:
We could exclude this callout since using `mintNextAndAddAuthMethods` with `sendPkpToItself` enabled, and the only Auth Method being the LA is effectively the same thing as MGB